### DISCUSSION TITLE
Instant Battery Charge

### DISCUSSION THREAD

**Discussion by Breseb** [mletenay/home-assistant-goodwe-inverter#307] [Ideas]
In the installer section of SolarGo you have the opertunity start an Instant Full charge of the battery. It would be nice to be able to start this via HomeAssistan. E.g. use cheap energy inputs at night.
![IMG_4325](https://github.com/user-attachments/assets/a8948a4f-586c-473e-a412-0bbb70ade9a2)

**Comment by janchlebek** [comment]
You can do this vie Eco Charge mode already - just set the mode and using the Eco Mode Power set the desired charging power

**Reply by Breseb** [reply]
i´m a bit confused. When i change to Eco-Mode in Homeassist, nothing happens with charging or discharging while looking into solarGo
![Unbenannt](https://github.com/user-attachments/assets/091d03d9-8a27-4310-8fdb-7958373018cc)

**Reply by Breseb** [reply]
Oh i got it. I had to configure the Eco-Mode (TOU) with SolarGo First: Add a new Slot. Time from 00:00-23:59 with 100% SOC and 100% Power on every weekday.
Now i only Need to Switch from normal Mode to eco and it starts loading.

Perfect thanks

**Reply by janchlebek** [reply]
> Oh i got it. I had to configure the Eco-Mode (TOU) with SolarGo First: Add a new Slot. Time from 00:00-23:59 with 100% SOC and 100% Power on every weekday. Now i only Need to Switch from normal Mode to eco and it starts loading.
> 
> Perfect thanks

You do not really need to set anything in the SolarGo app - there are two other modes than Eco Mode for your needs:

- Eco charge mode 
- Eco discharge mode

They do what it says. 

On top there is a numeric value Eco mode power that sets the power of dis/charge in percent of the inverter rated power.

You can then create automations based on any kind of parameters or time and charge or discharge the battery. To return to normal operation just set it to General mode.

**Reply by Breseb** [reply]
But how do i get these two modes? I only See these ![image](https://github.com/user-attachments/assets/2a83001c-7316-457c-af01-a0aea3a18ec6)

**Comment by janchlebek** [comment]
Strange - which inverter do you have? Are you using the official integration or the experimental from HACS?

**Reply by Breseb** [reply]
I have the GW5K-ET and use the normal Goodwe Integration. No experimental i think.

**Reply by janchlebek** [reply]
> I have the GW5K-ET and use the normal Goodwe Integration. No experimental i think.

This might be the reason - the Experimental has more features. Try to remove the official one and install Experimental via HACS
https://github.com/mletenay/home-assistant-goodwe-inverter?tab=readme-ov-file#differences-between-this-hacs-and-native-ha-integration

**Reply by Breseb** [reply]
Perfect, this works for me.
Now i have all the buttons you talked about.
Thanks very much.
